internal/dataplane/translator/translator.go

```diff
@@ -414,10 +415,24 @@ func (t *Translator) getUpstreams(serviceMap map[string]kongstate.Service) ([]ko
 					}
 				}
 
-				// add the new targets to the existing pool of targets for the Upstream.
-				targets = append(targets, newTargets...)
+				for _, t := range newTargets {
+					// See https://github.com/Kong/kubernetes-ingress-controller/issues/5761:
+					// Duplicate targets will appear in configurations that use Services with the same selector, which are used
+					// by some rollout systems. We need to deduplicate them while honoring the total weight.
+					//
+					// Because kongstate.Target is a nested kong.Target and the target IP is also a field named Target, the
+					// key names are a bit silly.
+					if existing, ok := targetMap[*t.Target.Target]; ok {
```

Comment:
`targetMap` maps from ? (`string`)  to ... IPs ? I'd be inclined to say that naming ( I know ) might be worth improving here.

Perhaps put `*.t.Target.Target` in a named variable? `ip`? `targetIP ?`

Reply:
Blame Kong upstream. The latest generated docs are unfortunately still kind of a mess, but it's literally just "target": https://docs.konghq.com/gateway/3.4.x/admin-api/#target-object

`target` is a hostname _or_ IP and its port, delimited by a colon, e.g. `example.com:80` or `10.0.01:9000` (I assume IPv6 uses `[fe80::1]:80` but didn't confirm). Ultimately it's _some_ unique string, and we trust that the endpoint resolver will give us a string that's a valid Kong target. We don't care about the details as long as it's unique. Will add a bit more to the comment.